Ginger/Ginger/ApplicationModelsLib/ModelParams Pages/ModelParamsPage.xaml.cs

```diff
@@ -177,6 +193,65 @@ private void AddGlobalParam(object sender, RoutedEventArgs e)
+            {
+                messageResult = System.Windows.MessageBox.Show("Do you want to delete also nodes from request body that contain those parameters?", "Delete nodes from request body?", System.Windows.MessageBoxButton.YesNoCancel, System.Windows.MessageBoxImage.Question, System.Windows.MessageBoxResult.No);
+            }
+
+            if (messageResult == System.Windows.MessageBoxResult.Yes)
+            {
+                OnParamDeleteEvent(new List<AppModelParameter>(ModelParametersGrid.Grid.SelectedItems.Cast<AppModelParameter>().ToList()));
+            }
+            else if(messageResult == System.Windows.MessageBoxResult.No)
+            {
+                if (ModelParametersGrid.Grid.SelectedItems.Count == 0)
+                {
+                    Reporter.ToUser(eUserMsgKeys.SelectItemToDelete);
+                    return;
+                }
+
+                ModelParametersGrid.DataSourceList.SaveUndoData();
+
+                List<object> SelectedItemsList = ModelParametersGrid.Grid.SelectedItems.Cast<object>().ToList();
+
+                foreach (object o in SelectedItemsList)
+                {
+                    ModelParametersGrid.DataSourceList.Remove(o);
+                }
+            }
+        }
+
+        private void ClearAllParams_Clicked(object sender, RoutedEventArgs e)
+        {
+            MessageBoxResult messageResult = System.Windows.MessageBoxResult.No;
+            if ((((ApplicationAPIModel)mAAMB).ContentType == ApplicationAPIUtils.eContentType.XML || ((ApplicationAPIModel)mAAMB).ContentType == ApplicationAPIUtils.eContentType.JSon))
+            {
+                messageResult = System.Windows.MessageBox.Show("Do you want to delete also nodes from request body that contain those parameters?", "Delete nodes from request body?", System.Windows.MessageBoxButton.YesNoCancel, System.Windows.MessageBoxImage.Question, System.Windows.MessageBoxResult.No);
+            }
+
+            if (messageResult == System.Windows.MessageBoxResult.Yes)
+            {
+                OnParamDeleteEvent(new List<AppModelParameter>(ParamsList));
+            }
+            else if (messageResult == System.Windows.MessageBoxResult.No)
+            {
+                if (ModelParametersGrid.Grid.Items.Count == 0)
+                {
+                    Reporter.ToUser(eUserMsgKeys.NoItemToDelete);
+                    return;
+                }
+
+                if ((Reporter.ToUser(eUserMsgKeys.SureWantToDeleteAll)) == MessageBoxResult.Yes)
+                {
+                    ModelParametersGrid.DataSourceList.SaveUndoData();
+                    ParamsList.ClearAll();
+                }
+            }
+        }
+
         private void AddGlobalParametertoAPIGlobalParameterList(ObservableList<GlobalAppModelParameter> APIGlobalParamList, GlobalAppModelParameter GAMP)
```

Comment:
we have duplicated code in DeleteParams_Clicked & ClearAllParams_Clicked, those 2 events handlers needs to call to function which will do the similar code

Reply:
Fixed